src/SignalR/common/Protocols.NewtonsoftJson/src/Protocol/NewtonsoftJsonHubProtocol.cs

```diff
@@ -348,12 +348,12 @@ private HubMessage ParseMessage(Utf8BufferTextReader textReader, IInvocationBind
                     case HubProtocolConstants.StreamItemMessageType:
                         if (itemToken != null)
                         {
-                            var returnType = binder.GetReturnType(invocationId);
                             try
                             {
-                                item = itemToken.ToObject(returnType, PayloadSerializer);
+                                var itemType = binder.GetStreamItemType(invocationId);
```

Comment:
Do we have any tests that cover the case when this throws a KeyNotFoundException?

Reply:
The `UploadStreamItemInvalidId` test is doing this, I'll take a look at our json protocols tests and add one if it's missing